combat-trainer.lic

```diff
@@ -2567,7 +2568,7 @@ class TrainerProcess
     when /^Collect$/i
       game_state.sheath_whirlwind_offhand
       retreat
-      put('engage') unless game_state.npcs.empty? || game_state.retreating?
+      put('engage') unless game_state.npcs.empty? || game_state.retreating? || !game_state.is_engaging_allowed?
```

Comment:
What do you think about making a function on game_state (`engage`?) that encapsulates this repeated set of conditionals? See lines 2621, 2812, , 3358, etc.

Reply:
Yeah, I thought about that. Also considered making `is_engaging_allowed?` defined as "not cast Innocence and not retreating"

There are some cases where we'll completely skip some actions when retreating that we should be able to do while using Innocence, so they're not quite the same thing, but there is a lot of repeat/overlap.

Maybe it would make sense to also include `game_state.npcs.empty?` in the check too. In that case, it doesn't _hurt_ anything to issue a `face` or `engage` command but it's still pointless.

Reply:
This could also address a number of the other concerns. "Why would an empath..." "Why would someone using Innocence..."

Generalizes more to "prevent us from issuing a `face` or `engage` command when it would be either bad or wrong"